client/memory.cpp

```diff
@@ -84,6 +92,27 @@ clEnqueueFillBuffer(cl_command_queue command_queue, cl_mem buffer,
 	}
 }
 
+SO_EXPORT CL_API_ENTRY void* CL_API_CALL
+clEnqueueMapBuffer(cl_command_queue command_queue, cl_mem buffer, cl_bool blocking_map,
+                   cl_map_flags map_flags, size_t offset, size_t size,
+                   cl_uint num_events_in_wait_list, const cl_event* event_wait_list,
+                   cl_event* event, cl_int *errcode_ret) CL_API_SUFFIX__VERSION_1_0
+{
+	void* ptr = std::malloc(size);
```

Comment:
This is a C-style allocation. This should be done either as a std::vector::resize of new[] into a std::unique_ptr.